Subject: Sync ext/hash installed headers

This syncs ext/hash headers installation on *nix and Windows:
- php_hash_joaat.h and php_hash_fnv.h added also on Windows installation
- xxhash/xxhash.h added on both installations as it is included in
  php_hash_xxhash.h
- Include path for xxhash.h changed to relative so the php_hash_xxhash.h
  can be included outside of php-src
- Redundant include flags removed

See: https://github.com/php/php-src/pull/13210
---
 ext/hash/config.m4         | 3 +--
 ext/hash/config.w32        | 8 ++------
 ext/hash/php_hash_xxhash.h | 2 +-
 3 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/ext/hash/config.m4 b/ext/hash/config.m4
index dd0b24a132..ff79668780 100644
--- a/ext/hash/config.m4
+++ b/ext/hash/config.m4
@@ -31,7 +31,6 @@ else
 fi
 
 PHP_ADD_BUILD_DIR(ext/hash/murmur, 1)
-PHP_HASH_CFLAGS="$PHP_HASH_CFLAGS -I@ext_srcdir@/xxhash"
 
 EXT_HASH_SOURCES="hash.c hash_md.c hash_sha.c hash_ripemd.c hash_haval.c \
   hash_tiger.c hash_gost.c hash_snefru.c hash_whirlpool.c hash_adler32.c \
@@ -41,7 +40,7 @@ EXT_HASH_HEADERS="php_hash.h php_hash_md.h php_hash_sha.h php_hash_ripemd.h \
   php_hash_haval.h php_hash_tiger.h php_hash_gost.h php_hash_snefru.h \
   php_hash_whirlpool.h php_hash_adler32.h php_hash_crc32.h \
   php_hash_fnv.h php_hash_joaat.h php_hash_sha3.h php_hash_murmur.h \
-  php_hash_xxhash.h"
+  php_hash_xxhash.h xxhash/xxhash.h"
 
 PHP_NEW_EXTENSION(hash, $EXT_HASH_SOURCES, 0,,$PHP_HASH_CFLAGS)
 PHP_INSTALL_HEADERS(ext/hash, $EXT_HASH_HEADERS)
diff --git a/ext/hash/config.w32 b/ext/hash/config.w32
index d38b032efd..1d0ca1e35c 100644
--- a/ext/hash/config.w32
+++ b/ext/hash/config.w32
@@ -34,13 +34,9 @@ if (!CHECK_HEADER_ADD_INCLUDE('PMurHash.h', 'CFLAGS_HASH', hash_murmur_dir)) {
 }
 ADD_SOURCES(hash_murmur_dir, 'PMurHash.c PMurHash128.c', 'hash');
 
-var hash_xxhash_dir = 'ext/hash/xxhash';
-if (!CHECK_HEADER_ADD_INCLUDE('xxhash.h', 'CFLAGS_HASH', hash_xxhash_dir)) {
-	ERROR('Unable to locate xxhash headers');
-}
-
 PHP_INSTALL_HEADERS('ext/hash/',	'php_hash.h php_hash_md.h php_hash_sha.h ' +
 									'php_hash_ripemd.h php_hash_haval.h php_hash_tiger.h ' +
 									'php_hash_gost.h php_hash_snefru.h php_hash_whirlpool.h ' +
 									'php_hash_adler32.h php_hash_crc32.h php_hash_sha3.h ' +
-									'php_hash_murmur.h php_hash_xxhash.h');
+									'php_hash_murmur.h php_hash_xxhash.h php_hash_fnv.h ' +
+									'php_hash_joaat.h xxhash/xxhash.h');
diff --git a/ext/hash/php_hash_xxhash.h b/ext/hash/php_hash_xxhash.h
index a1e8840ce2..ace70deedb 100644
--- a/ext/hash/php_hash_xxhash.h
+++ b/ext/hash/php_hash_xxhash.h
@@ -18,7 +18,7 @@
 #define PHP_HASH_XXHASH_H
 
 #define XXH_INLINE_ALL 1
-#include "xxhash.h"
+#include "xxhash/xxhash.h"
 
 typedef struct {
 	XXH32_state_t s;
